### PR TITLE
fix: fix wrong path for completion scripts

### PR DESCRIPTION
**Pull request by Jisu-Woniu**
- **fix: fix wrong path for completion scripts**

Fix issue referenced in <https://github.com/Enter-tainer/typstyle/pull/131#issuecomment-2365456088>

### CODE REVIEW

**Comment by Enter-tainer** [comment]
how does tinymist solve this problem?

**Comment by Jisu-Woniu** [comment]
Just my coding mistake.

**Comment by Enter-tainer** [comment]
https://github.com/Myriad-Dreamin/tinymist/blob/main/crates/tinymist/src/args.rs#L49

**Comment by Jisu-Woniu** [comment]
I have fixed it locally, and will squash the commits later.

**Comment by Enter-tainer** [comment]
can we copy-paste tinymist's impl?

**Comment by Jisu-Woniu** [comment]
Their implementation seems much too complicated for me 😔.

**Comment by Jisu-Woniu** [comment]
Their implementation seems much too complicated for me 😔.

**Comment by Jisu-Woniu** [comment]
They seem to be detecting shell from the `$SHELL` variable, but I consider it wrong and misleading since the variable is not related to the current running shell.

File: src/main.rs [hunk 1]
```diff
@@ -114,10 +114,22 @@ fn execute(args: CliArguments) -> Result<FormatStatus> {
...
+                let bin_name = bin_path
+                    .as_ref()
+                    .and_then(|p| p.file_stem().and_then(|p| p.to_str()))
+                    .unwrap_or("typstyle");
+
                 clap_complete::generate(
                     *shell,
                     &mut cli::CliArguments::command(),
-                    std::env::args().next().unwrap(),
+                    bin_name,
```
Comment (Enter-tainer) [inline src/main.rs:132]: maybe we should simply hardcode `"typstyle"` here?

Comment (Jisu-Woniu) [reply src/main.rs:132]: Just in case someone symlink'ed it.

File: src/main.rs [hunk 2]
```diff
@@ -114,10 +114,22 @@ fn execute(args: CliArguments) -> Result<FormatStatus> {
             #[cfg(feature = "completion")]
             cli::Command::Completions { shell } => {
                 use clap::CommandFactory;
+                use std::env::{args_os, current_exe};
+
+                let bin_path = args_os()
+                    .next()
+                    .map(PathBuf::from)
+                    .or_else(|| current_exe().ok());
```
Comment (Enter-tainer) [inline src/main.rs:122]: why should we use both args and current exe?

Comment (Enter-tainer) [reply src/main.rs:122]: I feel like using `"typstyle"` directly is much more easlier

Comment (Jisu-Woniu) [reply src/main.rs:122]: I see, symlink is beyond the scope of our consideration. The users can recompile or replace text with `sed` if they need to change this.

**Review by Enter-tainer — APPROVED** [review]

**Comment by Enter-tainer** [comment]
does this affect aur packaging? should i release a new version right now? i suppose it wont be a urgernt fix.

**Comment by Jisu-Woniu** [comment]
Yes, but maybe you can re-tag it with `git tag --force`, since I didn't push anything to AUR due to this error.

**Comment by Enter-tainer** [comment]
I've pushed v0.11.34

**Pull request merged**